sdk/storage/azure-storage-quickquery/src/main/java/com/azure/storage/quickquery/BlobQuickQueryClient.java

```diff
@@ -4,10 +4,142 @@
+import reactor.core.publisher.Mono;
+
+import java.io.IOException;
+import java.io.OutputStream;
+import java.io.UncheckedIOException;
+import java.nio.ByteBuffer;
+import java.time.Duration;
+
+import static com.azure.storage.common.implementation.StorageImplUtils.blockWithOptionalTimeout;
 
 /**
- * TEMP
+ * This class provides a client that contains all operations that apply to Azure Storage Blob quick query.
+ *
+ * <p>This client offers the ability to query contents of a structured blob. </p>
+ *
+ * @see BlobQuickQueryClientBuilder
  */
 @ServiceClient(builder = BlobQuickQueryClientBuilder.class)
 public class BlobQuickQueryClient {
+    private final ClientLogger logger = new ClientLogger(BlobQuickQueryClient.class);
+
+    private BlobQuickQueryAsyncClient client;
+
+    /**
+     * Package-private constructor for use by {@link BlobQuickQueryClientBuilder}.
+     * @param client the async blob quick query client
+     */
+    BlobQuickQueryClient(BlobQuickQueryAsyncClient client) {
+        this.client = client;
+    }
+
+    /**
+     * Opens a blob input stream to query the blob.
+     * <p><strong>Code Samples</strong></p>
+     *
+     * {@codesnippet com.azure.storage.quickquery.BlobQuickQueryClient.openInputStream#String}
+     *
+     * @param expression The query expression.
+     */
+    public final BlobQuickQueryInputStream openInputStream(String expression) {
+        return openInputStream(expression, null, null, null, null, null);
+    }
+
+    /**
+     * Opens a blob input stream to query the blob.
+     *
+     * <p><strong>Code Samples</strong></p>
+     *
+     * {@codesnippet com.azure.storage.quickquery.BlobQuickQueryClient.openInputStream#String-BlobQuickQuerySerialization-BlobQuickQuerySerialization-BlobRequestConditions-ErrorReceiver-ProgressReceiver}
+     *
+     * @param expression The query expression.
+     * @param input {@link BlobQuickQuerySerialization Serialization input}.
+     * @param output {@link BlobQuickQuerySerialization Serialization output}.
+     * @param requestConditions {@link BlobRequestConditions}
+     * @param nonFatalErrorReceiver {@link ErrorReceiver<BlobQuickQueryError>}
+     * @param progressReceiver {@link ProgressReceiver}
+     */
+    public final BlobQuickQueryInputStream openInputStream(String expression, BlobQuickQuerySerialization input,
+        BlobQuickQuerySerialization output, BlobRequestConditions requestConditions,
+        ErrorReceiver<BlobQuickQueryError> nonFatalErrorReceiver, ProgressReceiver progressReceiver) {
+
+        // Data to subscribe to and read from.
+        Flux<ByteBuffer> data = client.queryWithResponse(expression, input, output, requestConditions)
+            .flatMapMany(ResponseBase::getValue);
+
+        // Create input stream from the data.
+        FluxInputStream fluxInputStream = new FluxInputStream(data, logger);
+
+        return new BlobQuickQueryInputStream(fluxInputStream, nonFatalErrorReceiver, progressReceiver);
```

Comment:
Why a progress receiver? Typically we only take that when we're doing parallel operations

Reply:
The service returns "progress" records of how many bytes of the original blob have been parsed so far, so we wanted to expose this to the user as theyre reading from the stream